src/main/java/uk/ac/ebi/embl/gff3tools/gff3/GFF3Annotation.java

```diff
@@ -121,13 +121,11 @@ public String getAccession() {
         }
     }
 
-    public List<GFF3Feature> getFeaturesByName(String featureName) {
-        return features.stream()
-                .filter(ftr -> ftr.getName().equalsIgnoreCase(featureName))
-                .collect(Collectors.toList());
-    }
-
     public List<GFF3Feature> getAllFeatures() {
         return Collections.unmodifiableList(this.features);
     }
+
+    public void removeFeature(GFF3Feature feature) {
+        features.remove(feature);
```

Comment:
Not for now, but I'm thinking this method may be slow on large annotations. If we need in the future we can optimize this by doing a binary search (features are ordered by location) and deleting by index. This would reduce the complexity to LogN.